From 02705a3a9cebc1dbb831f20946db6c5bc2c219c3 Mon Sep 17 00:00:00 2001
From: Christophe Kerello <christophe.kerello@foss.st.com>
Date: Wed, 19 Feb 2025 18:09:16 +0100
Subject: [PATCH 1543/1695] dts: stm32: add fixed IO compensation on vddio
 regulators for stm32mp251

IO compensation could alter duty cycle of high frequency output signal
on stm32mp25 SoC. To avoid such issue, fixed IO compensation is used.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 7191becdce8a81b28cbb4314bdcd2fd31dc38fe0 dts: stm32: add fixed IO compensation on vddio regulators for stm32mp251]
Change-Id: I574974eb6c8fc304294f0d85d96d188aab318642
Signed-off-by: Christophe Kerello <christophe.kerello@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/438578
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Christophe KERELLO <christophe.kerello@st.com>
Tested-by: Christophe KERELLO <christophe.kerello@st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/arch/arm/dts/stm32mp251.dtsi | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp251.dtsi b/core/arch/arm/dts/stm32mp251.dtsi
index df94faf3c..2c16df453 100644
--- a/core/arch/arm/dts/stm32mp251.dtsi
+++ b/core/arch/arm/dts/stm32mp251.dtsi
@@ -940,30 +940,35 @@
 			vddio1: vddio1 {
 				compatible = "st,stm32mp25-pwr-regu";
 				regulator-name = "vddio1";
+				st,iocomp = <8 7>;
 				status = "disabled";
 			};
 
 			vddio2: vddio2 {
 				compatible = "st,stm32mp25-pwr-regu";
 				regulator-name = "vddio2";
+				st,iocomp = <8 7>;
 				status = "disabled";
 			};
 
 			vddio3: vddio3 {
 				compatible = "st,stm32mp25-pwr-regu";
 				regulator-name = "vddio3";
+				st,iocomp = <8 7>;
 				status = "disabled";
 			};
 
 			vddio4: vddio4 {
 				compatible = "st,stm32mp25-pwr-regu";
 				regulator-name = "vddio4";
+				st,iocomp = <8 7>;
 				status = "disabled";
 			};
 
 			vddio: vddio {
 				compatible = "st,stm32mp25-pwr-regu";
 				regulator-name = "vddio";
+				st,iocomp = <8 7>;
 				regulator-always-on;
 				status = "disabled";
 			};
-- 
2.47.2

